virttest/test_setup.py

```diff
@@ -513,8 +513,8 @@ def set_hugepages(self):
             except IOError:
                 msg = "Can't read/write from kernel hugepage file"
                 raise exceptions.TestSetupFail(msg)
-            if loop_hp == hp:
-                raise ValueError("Cannot set the kernel hugepage setting "
+            if hp != self.target_hugepages:
+                raise exceptions.TestCancel("Cannot set the kernel hugepage setting "
```

Comment:
This would be a TestSetupFail(), also please fix indentation in below 2 lines for travis.